great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py

```diff
@@ -422,10 +435,26 @@ def get_metrics(
             metric_configurations=metrics_to_resolve
         )
 
-        # Step-5: Map resolved metrics to their attributes for identification and recovery by receiver.
+        # Step-6: Sort resolved metrics according to same sort order as was applied to "MetricConfiguration" directives.
+
+        resolved_metrics_sorted: Dict[Tuple[str, str, str], Any] = {}
```

Comment:
I'd like to consider introducing a type that inherits from Tuple[str, str, str] that can represent a metric reference; this might be a way of improving the readability of the code base without a major refactor (and could be introduced incrementally)